vllm/distributed/kv_transfer/vllm_adapter.py

```diff
@@ -71,82 +73,98 @@ def __init__(
+                self.send_signal_pipe = TorchDistributedPipe(
+                    self.group_ranks,
+                    self.local_rank,
+                    "gloo",
+                )
+                self.recv_pipe = TorchDistributedPipe(
+                    self.group_ranks,
+                    self.local_rank,
+                    DISTRIBUTED_BACKEND,
+                )
+                self.recv_signal_pipe = TorchDistributedPipe(
+                    self.group_ranks,
+                    self.local_rank,
+                    "gloo",
+                )
+                self.sender = SimpleKVLookupBuffer(self.send_signal_pipe,
+                                                        self.send_pipe,
+                                                        self.lookup_buffer_size)
+                self.recver = SimpleKVLookupBuffer(self.recv_signal_pipe,
+                                                        self.recv_pipe,
+                                                        self.lookup_buffer_size)
+                self.tensor_device = DISTRIBUTED_DEVICE
+            else:
+
+                # the current vLLM instance is KV consumer, so it needs to connect
+                # its recv pipe to the send pipe of KV producder
+
+                self.recv_pipe = TorchDistributedPipe(
+                    self.group_ranks,
+                    self.local_rank,
+                    DISTRIBUTED_BACKEND,
+                )
+                self.recv_signal_pipe = TorchDistributedPipe(
+                    self.group_ranks,
+                    self.local_rank,
+                    "gloo",
+                )
+                self.send_pipe = TorchDistributedPipe(
+                    self.group_ranks,
+                    self.local_rank,
+                    DISTRIBUTED_BACKEND,
+                )
+                self.send_signal_pipe = TorchDistributedPipe(
+                    self.group_ranks,
+                    self.local_rank,
+                    "gloo",
+                )
+                self.sender = SimpleKVLookupBuffer(self.send_signal_pipe,
+                                                        self.send_pipe,
+                                                        self.lookup_buffer_size)
+                self.recver = SimpleKVLookupBuffer(self.recv_signal_pipe,
+                                                        self.recv_pipe,
+                                                        self.lookup_buffer_size)
+                self.tensor_device = DISTRIBUTED_DEVICE
+        elif self.kv_transfer_driver.startswith("valkey"):
+            url = self.kv_transfer_driver.split("://")[1]
+            ip, port = parse_url(url)
+            # TODO add PING command
+            self.sender = KVDatabaseTransfer(ip, int(port), self.local_rank, ValkeyPipe())
+            self.recver = KVDatabaseTransfer(ip, int(port), self.local_rank, ValkeyPipe())
+        else:
+            raise ValueError("Invalid kv_transfer_driver")
```

Comment:
lol we definitely need a factory class to build the lookup buffer in the future, but let us keep it as is for now.